### DISCUSSION TITLE
DietPi Conversion from a specific debian image?

### DISCUSSION THREAD

**Discussion by KalixtShawxo** [MichaIng/DietPi#5076] [Q&A] [answered]
I am wanting to setup a FarmPi  (OctoFarm on a PI) using DietPi. to support my two OctoPrint servers for my 3D printers. 
The problem I run into is that it relies on MongoDB which does not yet support Bullseye.
So I was wondering if i download/install the image https://kevenaar.name/farmpi-running-octofarm-on-a-raspberry-pi/
(OctoFarm on a premade PI image using Ubuntu 64) and then ran the DietPi conversion script on it ....
would the existing software still work or would DietPi being Bullseye based break the Buster based operation?
Hardware is a regular raspberry Pi 3 B with a touchscreen PiTFT on it. 
I want to use DietPi so that I can easily set it up without a DE with chromium kiosk and have the browser just start on the webpage for the OCtoFarm. 

Am I being to ambitious for a newb? 

Any help or information/guidance would be greatly appreciated.  Thanks!

**Comment by Joulinar** [comment]
Hi,

DietPi is based on Debian. Therefore you can't use an Ubuntu image as source. Furthermore, on converting an image into DietPi you would loose all preinstalled software because our PREP script would remove everything not needed for a DietPi base installation. This include MongoDB. 

As an alternative, you could try using Docker to run MongoDB https://github.com/MichaIng/DietPi/issues/4799#issuecomment-934948877

**Comment by KalixtShawxo** [comment]
ok. I thought ubuntu was a debian variant and could still work.  now I know. 
I'll look into the Docker setup, thank you very much for the direction.

**Comment by MichaIng** [comment, marked as answer]
Since we have RPi images, this is not an image request. I think we should make more clear how we intend GitHub discussions to be used currently. Or shall we widen it to include a Q&A category (where this fits)?

> I thought ubuntu was a debian variant

It is based on Debian, i.e. uses 99% of the Debian package sources, but it ships own package builds with own APT repositories which are not compatible with the Debian ones. I once tried to convert an Ubuntu image to Debian/DietPi. It somewhat worked with some manual hacks, but it cannot be done/used as a reliable basis for end users 🙂: https://github.com/MichaIng/DietPi/issues/1557

**Comment by KalixtShawxo** [comment]
Thanks again for the help and the clarification, I apologize if my request is outside the scope of the discussion requirements. 
I am too new to understand the docker discussion you referenced. So I may have to abandon the idea of FarmPi under DietPi for now, until I can educate myself a bit more. Still loving DietPi for my other projects. Overall it is very welcoming to someone new like me. 

Sent from Yahoo Mail on Android 
 
  On Tue, Dec 14, 2021 at 10:50, ***@***.***> wrote:   


Since we have RPi images, this is not an image request. I think we should make more clear how we intend GitHub discussions to be used currently. Or shall we widen it to include a Q&A category (where this fits)?


I thought ubuntu was a debian variant


It is based on Debian, i.e. uses 99% of the Debian package sources, but it ships own package builds with own APT repositories which are not compatible with the Debian ones. I once tried to convert an Ubuntu image to Debian/DietPi. It somewhat worked with some manual hacks, but it cannot be done/used as a reliable basis for end users 🙂 : #1557

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
